pkg/controller/pod_volume_restore_controller.go

```diff
@@ -220,10 +220,11 @@ func isPodOnNode(pod *corev1api.Pod, node string) bool {
 }
 
 func isResticInitContainerRunning(pod *corev1api.Pod) bool {
-
 	// Restic wait container can be anywhere in the list of init containers, but must be running.
 	i := getResticInitContainerIndex(pod)
-	return i >= 0 && pod.Status.InitContainerStatuses[i].State.Running != nil
+	return i >= 0 &&
+		len(pod.Status.InitContainerStatuses) > 0 &&
```

Comment:
Would it better by checking `len(pod.Status.InitContainerStatuses)-1 >= i`? It's more robust to make sure the index won't out of range. 

Also, it's good to have a test case for this.

Reply:
Ah, that's a very good catch, I'm sorry I missed it.

Reply:
@jenting Thanks for kind recommendation. But I think `len(pod.Status.InitContainerStatuses)-1 >= i` may incur runtime error.

Counterexample: multiple InitContainers but no restic InitContainer 
- Multiple InitContainers make `len(pod.Status.InitContainerStatuses)-1` larger than 0.
- No restic InitContainer makes `i` to `-1`, which is returned by `getResticInitContainerIndex`.
- As a result, `len(pod.Status.InitContainerStatuses)-1 >= i` becomes `true` and code executes to check next conditional, `pod.Status.InitContainerStatuses[i].State.Running != nil`.

Then, runtime error can occur because it accesses -1 index of `pod.Status.InitContainerStatuses`.

Reply:
> No restic InitContainer makes i to -1, which is returned by getResticInitContainerIndex.

So, the `i >= 0` would be _false_, then I think the code won't execute the `len(pod.Status.InitContainerStatuses)-1 >= i` 🤔  ?

Reply:
> So, the i >= 0 would be false,

Oh I misunderstood you suggest to remove `i >= 0` either.
Yes then, `len(pod.Status.InitContainerStatuses)-1 >= i` is more robust than `len(pod.Status.InitContainerStatuses) > 0` with `i >= 0`.
Modifying that line.